src/product/dto/create-product.dto.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  IsIn,
+  IsInt,
+  IsNumber,
+  IsPositive,
+  IsString,
+  Min,
+} from 'class-validator';
+import { ProductState } from '../types/product';
+
+export class CreateProductDto {
+  @IsString()
+  name: string;
+
+  @IsString()
+  description: string;
+
+  @IsNumber()
+  @IsInt()
+  @Min(0)
+  num_available_items: number;
+
+  @IsNumber()
+  @IsPositive()
+  unit_price: number;
+
+  @IsString()
+  @IsIn(Object.values(ProductState))
```

Comment:
In 'class-validator' we have available the `IsEnum` decorator too, have you tried it?
```suggestion
  @IsEnum(ProductState)
```

Reply:
Yes, I applied this change as discussed above